awx/main/credential_plugins/hashivault.py

```diff
@@ -177,7 +177,7 @@ def handle_auth(**kwargs):
         token = method_auth(**kwargs, auth_param=approle_auth(**kwargs))
     elif kwargs.get('kubernetes_role'):
         token = method_auth(**kwargs, auth_param=kubernetes_auth(**kwargs))
-    elif kwargs.get('username') and kwargs.get('password'):
+    elif kwargs.get('username') or kwargs.get('password'):
         token = method_auth(**kwargs, auth_param=ldap_auth(**kwargs))
```

Comment:
With this change to `or` - what if only one is given? (specifically when `ldap_auth()` is called)

Reply:
Thank you, Rick! I actually hadn't meant to leave this as `or` I was testing to see if it had an impact on the bad gateway error I keep getting, which it did not. I will get this changed!